raft/clock.go

```diff
@@ -39,8 +40,11 @@ func NewClock() *Clock {
 // AfterApplyInterval returns a channel that fires after the apply interval.
 func (c *Clock) AfterApplyInterval() <-chan chan struct{} { return newClockChan(c.ApplyInterval) }
 
-// AfterElectionTimeout returns a channel that fires after the election timeout.
-func (c *Clock) AfterElectionTimeout() <-chan chan struct{} { return newClockChan(c.ElectionTimeout) }
+// AfterElectionTimeout returns a channel that fires after a duration that is
+// between the election timeout and double the election timeout.
```

Comment:
This matches the Raft paper. If all the nodes return the same timeout then they tend to get stuck in an election loop indefinitely where they can't get enough votes.